BossMod/QuestBattle/QuestBattleWindow.cs

```diff
@@ -0,0 +1,208 @@
+    private readonly AbandonDuty abandonDutyHook = Marshal.GetDelegateForFunctionPointer<AbandonDuty>(Service.SigScanner.ScanText("E8 ?? ?? ?? ?? 41 B2 01 EB 39"));
+
+    public QuestBattleWindow(QuestBattleDirector director) : base(_windowID, false, new(300, 200), ImGuiWindowFlags.NoScrollbar | ImGuiWindowFlags.NoCollapse)
+    {
+        _director = director;
+        _director.World.CurrentZoneChanged.Subscribe(OnZoneChange);
+    }
+
+    private void OnZoneChange(WorldState.OpZoneChange zc)
+    {
+        Waymarks.Clear();
+    }
+
+    public override void PreOpenCheck()
+    {
+        IsOpen = _director.Enabled && _config.ShowUI;
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+            _director.Dispose();
+        base.Dispose(disposing);
+    }
+
+    public override void Draw()
+    {
+        if (_director.CurrentModule is QuestBattle qb)
+        {
+            ImGui.Text($"Module: {qb.GetType().Name}");
+            DrawObjectives(qb);
+
+            ImGui.Spacing();
+            ImGui.Separator();
+            ImGui.Spacing();
+        }
+
+        ImGui.TextUnformatted($"Zone: {World.CurrentZone} / CFC: {World.CurrentCFCID}");
+#if DEBUG
+        ImGui.SameLine();
+        GenerateModule();
+#endif
+        if (World.Party.Player() is Actor player)
+        {
+            ImGui.TextUnformatted($"Position: {Utils.Vec3String(player.PosRot.XYZ())}");
+#if DEBUG
+            ImGui.SameLine();
+            if (ImGui.Button("Copy vec"))
+            {
+                var x = player.PosRot.X;
+                var y = player.PosRot.Y;
+                var z = player.PosRot.Z;
+                ImGui.SetClipboardText($"new Vector3({x:F2}f, {y:F2}f, {z:F2}f)");
+            }
+            ImGui.SameLine();
+            if (ImGui.Button("Copy moveto"))
+            {
+                var x = player.PosRot.X;
+                var y = player.PosRot.Y;
+                var z = player.PosRot.Z;
+                ImGui.SetClipboardText($"/vnav moveto {x:F2} {y:F2} {z:F2}");
+            }
+#endif
+            if (World.Actors.Find(player.TargetID) is Actor tar)
+            {
+                ImGui.TextUnformatted($"Target: {tar.Name} ({tar.Type}; {tar.OID:X}) (hb={tar.HitboxRadius})");
+                ImGui.TextUnformatted($"Distance: {player.DistanceToHitbox(tar)}");
+                ImGui.TextUnformatted($"Angle: {player.AngleTo(tar)}");
+            }
+
+#if DEBUG
+            ImGui.Spacing();
+            ImGui.Separator();
+            ImGui.Spacing();
+
+            if (ImGui.Button("Record position"))
+                Waymarks.Add(player.Position);
+
+            if (ImGui.Button("Copy all"))
+                ImGui.SetClipboardText(string.Join(", ", Waymarks.Select(w => $"new({w.X:F2}f, {w.Z:F2}f)")));
+
+            foreach (var w in Waymarks)
+                ImGui.TextUnformatted($"{w}");
+#endif
+        }
+    }
+
+#if DEBUG
+    private void GenerateModule()
```

Comment:
Can't it be all done in replay analysis, similar to how it's done for normal modules?

Reply:
this is way easier for me because it can be done while i'm in the duty and i can rebuild -> reload. the replay viewer is pretty clunky